### DISCUSSION TITLE
How to return error messages to client?

### DISCUSSION THREAD

**Discussion by awatson1978** [accounts-js/accounts#1208] [Q&A]
Hello,
Neither a bug nor a feature request (yet), just trying to figure out the recommended approach to returning an error message to the client.

When I try to log in using an incorrect password, I can generate an `AccountsJsError` easily enough, and I see the `Invalid credentials` message in the server console.
 
<img width="1753" alt="Screen Shot 2021-12-16 at 3 33 12 AM" src="https://user-images.githubusercontent.com/675910/146346286-cf1c8650-8f27-45cc-b1ad-0d5aae1f67f3.png">

However, when I try to catch the error on the client, I'm getting a TypeError, because the `accounts/password/authenticate` endpoint isn't returning a proper payload.  

<img width="1612" alt="Screen Shot 2021-12-16 at 3 33 29 AM" src="https://user-images.githubusercontent.com/675910/146346644-76b02d6b-82ca-457a-93a1-3156feeb67f6.png">

Here is my code.
<img width="1919" alt="Screen Shot 2021-12-16 at 3 32 20 AM" src="https://user-images.githubusercontent.com/675910/146346130-30ed79da-fb53-439f-ae99-15a4d506762b.png">

Should I be using a specific HTTP message code?  Does the error get put into the body of the response?  In the header?  Something else?  

Any guidance on how to wire this up correctly would be much appreciated!
-aw

**Comment by pradel** [comment]
What you could do for this:
- in your catch clause server-side replace the `code` property of the payload to the corresponding code property in the error object.
- in the client side do not check the AccountsError type but respect a JSON object with a `code` property.
